src/tlp_lib/smartcontracts/EthereumSC.py

```diff
@@ -45,8 +45,21 @@ def commitments(self) -> TLP_Digests:
 
     @commitments.setter
     def commitments(self, commitments: TLP_Digests):
-        if not self._has_succeeded(self._contract.functions.setCommitments(commitments)):
-            raise RuntimeError("Commitments were not set correctly")
+        total_commitments = len(commitments)
+        start_index = 0
+
+        while start_index < total_commitments:
```

Comment:
I don't think you need to do this manually, there's a function in the `itertools` module that helps with this: https://docs.python.org/3.12/library/itertools.html#itertools.batched

You can then iterate over the generator like

```py
for commitments_batch in itertools.batched(commitments, self._SC_PUZZLE_BATCH_SIZE):
```

Reply:
Thank you, did not know this existed, I have changed it accordingly

Reply:
yeah, the standard library has been getting a lot of useful stuff with the last few releases